air/src/air/coefficients.rs

```diff
@@ -28,12 +31,71 @@ use math::{get_power_series, get_power_series_with_offset, FieldElement};
 ///
 /// The coefficients are separated into two lists: one for transition constraints and another one
 /// for boundary constraints. This separation is done for convenience only.
+///
+/// Note that the soundness error of the protocol will depend on the batching used when computing
+/// the constraint composition polynomial. More precisely, when using algebraic batching there
+/// might be a loss of log_2(C - 1) bits of RbR soundness of the protocol, where C is the total
+/// number of constraints.
 #[derive(Debug, Clone)]
 pub struct ConstraintCompositionCoefficients<E: FieldElement> {
     pub transition: Vec<E>,
     pub boundary: Vec<E>,
 }
 
+impl<E: FieldElement> ConstraintCompositionCoefficients<E> {
+    /// Generates the random values used in the construction of the constraint composition
+    /// polynomial when linear batching is used.
+    pub fn draw_linear(
+        public_coin: &mut impl RandomCoin<BaseField = E::BaseField>,
+        num_of_transition_constraints: usize,
+        num_of_boundary_constraints: usize,
```

Comment:
nit: we usually simply say `num_transition_constraints` (without the `of` part).